int32 sonar1
int32 sonar2
int32 sonar3
int32 sonar4
